src/tools/language-server/diagnostics.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * @license
+ * Copyright (c) 2019 Google Inc. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {Diagnostic, DiagnosticSeverity, Range} from 'vscode-languageserver';
+
+import {Manifest, ManifestError} from '../../runtime/manifest.js';
+import {SourcePosition} from '../../runtime/manifest-ast-nodes.js';
+
+import {Logger, uri2path, normalizeUri} from './util.js';
+import {LspLoader} from './lspLoader.js';
+
+
+export async function gatherDiagnostics(uri: string, logger: Logger):
```

Comment:
I wasn't sure when reading this whether these diagnostics are for debugging/development, that is, are they IDE-user-facing? Or for you, or others, when working on LSP? Seems worth a file/class doc comment.